### PR TITLE
14244 Add Document Id for drafts

### PR DESCRIPTION
**Pull request by ChasBoucher**
*Issue #:* /bcgov/entity###14244

*Description of changes:*

- Add document id to child rows if not a draft

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

**Review by dimak1 — APPROVED** [review]
Could you bump up the version please.

File: ppr-ui/src/components/tables/common/TableRow.vue [hunk 1]
```diff
@@ -29,12 +29,13 @@
           </v-btn>
         </v-col>
         <v-col style="padding-top: 2px;">
-          <p v-if="isDraft(item)" :class="{ 'ma-0': true, 'pl-9': isChild }">Pending</p>
+          <p v-if="isDraft(item)" :class="{ 'ma-0': true, 'pl-9': isChild}">Pending</p>
```
Comment (cameron-eyds) [inline ppr-ui/src/components/tables/common/TableRow.vue]: Not a deal breaker but the space can stay, probably a misclick?

File: ppr-ui/src/components/tables/common/TableRow.vue [hunk 2]
```diff
@@ -29,12 +29,13 @@
...
           <!-- child drafts will sometimes show outside their base reg during the sort -->
           <div v-if="isChild || (isDraft(item) && item.baseRegistrationNumber)" :class="isChild ? 'pl-9' : ''">
-            <p class="ma-0">{{ item.registrationNumber }}</p>
+            <p v-if="!isPpr" style="font-size: 0.875rem !important;" :class="{ 'ma-0': true }">{{ item.documentId }}</p>
+            <p v-if="isPpr" class="ma-0">{{ item.registrationNumber }}</p>
```
Comment (cameron-eyds) [inline ppr-ui/src/components/tables/common/TableRow.vue]: Maybe a good chance to use `v-if` and `v-else` 
v-if isPpr / v-else you doc id. 

Also, your styling in the `!isPpr` is a little off. Your bound class is always `true`? 
Also, is the `important` required? `p` default should be .875rem so if something is overriding, perhaps make a class to keep the template neat.

File: ppr-ui/tests/unit/TableRow.spec.ts [hunk 3]
```diff
@@ -208,7 +208,7 @@ describe('TableRow tests', () => {
         expect(rowData.exists()).toBe(true)
         expect(rowData.length).toBe(11)
         // reg num
-        if (isChild) expect(rowData.at(0).text()).toBe(`Pending  Base Registration: ${baseReg.baseRegistrationNumber}`)
+        if (isChild) expect(rowData.at(0).text()).toBe(`Pending   Base Registration: ${baseReg.baseRegistrationNumber}`)
```
Comment (cameron-eyds) [inline ppr-ui/tests/unit/TableRow.spec.ts]: Additional spacing intentional?

**Review by cameron-eyds — COMMENTED** [review]
Just a few comments!

File: ppr-ui/src/components/tables/common/TableRow.vue (hunk 1 above)
Comment (ChasBoucher) [reply ppr-ui/src/components/tables/common/TableRow.vue]: Made the change...

File: ppr-ui/src/components/tables/common/TableRow.vue (hunk 2 above)
Comment (ChasBoucher) [reply ppr-ui/src/components/tables/common/TableRow.vue]: Good point..

File: ppr-ui/tests/unit/TableRow.spec.ts (hunk 3 above)
Comment (ChasBoucher) [reply ppr-ui/tests/unit/TableRow.spec.ts]: Yes... It takes care of the line break I added. 
The line break <br>was added to ensure there is a line break for both Ppr and registration number wraps.

Comment (cameron-eyds) [reply ppr-ui/tests/unit/TableRow.spec.ts]: Sounds good, thanks!

File: ppr-ui/src/components/tables/common/TableRow.vue [hunk 4]
```diff
@@ -29,11 +29,11 @@
           </v-btn>
         </v-col>
         <v-col style="padding-top: 2px;">
-          <p v-if="isDraft(item)" :class="{ 'ma-0': true, 'pl-9': isChild}">Pending</p>
+          <p v-if="isDraft(item)" :class="{ 'ma-0': true, 'pl-9': isChild }">Pending</p>
           <!-- child drafts will sometimes show outside their base reg during the sort -->
           <div v-if="isChild || (isDraft(item) && item.baseRegistrationNumber)" :class="isChild ? 'pl-9' : ''">
-            <p v-if="!isPpr" style="font-size: 0.875rem !important;" :class="{ 'ma-0': true }">{{ item.documentId }}</p>
             <p v-if="isPpr" class="ma-0">{{ item.registrationNumber }}</p>
+            <p v-else style="font-size: 0.875rem;" :class="{ 'ma-0': true }">{{ item.documentId }}</p>
```
Comment (cameron-eyds) [inline ppr-ui/src/components/tables/common/TableRow.vue]: This class binding is still set to always `true`. 

Using class binding, ie `:class={}` is a way to conditionally add a class. In this case it's always `true` so there is no condition. 
Was the intention here to apply a condition? 
If not, you can simply add the class to the element, ie `class="ma-0"`

Comment (cameron-eyds) [reply ppr-ui/src/components/tables/common/TableRow.vue]: @ChasBoucher

Comment (ChasBoucher) [reply ppr-ui/src/components/tables/common/TableRow.vue]: I am not sure why it is there... I did'nt change the existing code other than the condition... but I will make the change.

Comment (cameron-eyds) [reply ppr-ui/src/components/tables/common/TableRow.vue]: Fair enough, all good! 
Thank you 👍

**Review by cameron-eyds — APPROVED** [review]

**Review by owillborn — APPROVED** [review]
Looks solid

**Pull request merged**